### PR TITLE
Added Database checks

### PR DESCRIPTION
**Pull request by shivanshuag**
Issue #27 
### Size
- [x] Info - Returns the size of the database
- [x] Fail - If the sql query does not succeed for any reason
### Collation
- [x] Pass - Passes on default drupal installation
- [x] Warn - Collation of any table is `latin1_swedish_ci`
- [x] Info - Return collation of all the tables which are not utf8
### Engine
- [x] Pass - If the database engine of all tables is `InnoDB`
- [x] Fail - If the database engine for any table is not `InnoDB`
### RowCount
- [x] Info - No table has more than 1000 rows
- [x] Warn - If some table has more than 1000 rows

### CODE REVIEW

**Comment by fluxsauce** [comment]
@shivanshuag merge conflict

**Comment by fluxsauce** [comment]
Size Fail - does not catch exception.

**Comment by fluxsauce** [comment]
> Warn - Collation of any table is latin1_swedish_ci

Does not happen.

```
jon@Decker ~/projects/drupal-8.0.0-beta10 $ drush audit-database --detail
Database: 100%
  No action required.                                                                                                                   [success]
  Total size: Determine the size of the database.
    Total size: 9.38MB
  Collations: Check to see if there are any tables that aren't using UTF-8.
    Table Name: Collation
    ---------------------
    config: utf8_swedish_ci
  Storage Engines: Check to see if there are any tables that aren't using InnoDB.
    Every table is using InnoDB.
  Tables with at least 1000 rows: Return list of all tables with at least 1000 rows in the database.
    No tables with less than 1000 rows.
jon@Decker ~/projects/drupal-8.0.0-beta10 $ drush audit-database
Database: 100%
  No action required.
```

File: Check/Database/Engine.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+   */
+  public function getResultWarn() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getAction().
+   */
+  public function getAction() {
+    if ($this->score != SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_PASS) {
+      return dt('Change the Storage Engine to InnoDB. See @url for details.', array(
+        '@url' => 'http://dev.mysql.com/doc/refman/5.5/en/converting-tables-to-innodb.html',
```
Comment (fluxsauce) [inline Check/Database/Engine.php]: Link to current major version, not 5.5

**Comment by fluxsauce** [comment]
Please resolve merge conflict and outstanding action items, then remove the needs work label.

**Comment by shivanshuag** [comment]
What is the reason behind catching database exception in Size check and not any other check which performs an sql query?

**Comment by fluxsauce** [comment]
> What is the reason behind catching database exception in Size check and not any other check which performs an sql query?

Because if Size fails, everything else will fail. With that said, there's no reason not to catch them all.

**Comment by fluxsauce** [comment]
Good work, everything checks out! I added missing translation functions, probably were missing in the Drupal 7 version.

**Pull request merged**